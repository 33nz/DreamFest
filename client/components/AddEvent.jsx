import React from 'react'
import { addEvent } from '../api/events'
import { showError } from '../actions/error'
import { connect } from 'react-redux'

class AddEvent extends React.Component {
  state = {
    title: '',
    date: '',
    volunteers: 0,
    description: ''
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    const newEvent = {
      ...this.state,
      gardenId: this.props.gardenId
    }

    return addEvent(newEvent)
      .then(() => {
        this.props.history.push('/garden')
        return null
      })
      .catch((error) => {
        this.props.dispatch(showError(error.message))
      })
  }

  render () {
    return (
      <>
        <div className="column form">
          <form className='column is-two-fifths pl-0 is-pulled-left'>
            <h2 className='mb-6'>Create New Event</h2>
            <label className='label'>Event Title</label>
            <input
              className="input is-normal"
              type="text"
              placeholder="Event Title"
              name="title"
              value={this.state.title}
              onChange={this.handleChange}
            />

            <h5 className='label'>Date</h5>
            <input
              className="input is-normal"
              type="date"
              name="date"
              value={this.state.date}
              onChange={this.handleChange}
            />

            <label className='label'>Volunteers Needed</label>
            <input
              className="input is-normal"
              type="number"
              name="volunteers"
              value={this.state.volunteers}
              onChange={this.handleChange}
            />

            <label className='label'>Event Description</label>
            <textarea
              className="textarea is-normal"
              placeholder="Event Description"
              name="description"
              value={this.state.description}
              onChange={this.handleChange}
            />

            <button
              className="button mt-6"
              onClick={this.handleSubmit}
            >Create</button>
          </form>

          <div className="column is-two-fifths is-pulled-right">
            <div className='eventPreview'>
              <h1 className='mb-6'>Event Preview</h1>
              <div className="box mt-78">
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Sed quasi, harum omnis a non, voluptas quam ipsam possimus iure ipsa est, aut error doloribus saepe aliquam nostrum fuga! Aperiam, non.
              </div>
            </div>
          </div>
        </div>
      </>

    )
  }
}

const mapStateToProps = (state) => {
  return {
    gardenId: state.user.gardenId,
    error: state.error
  }
}

export default connect(mapStateToProps)(AddEvent)
