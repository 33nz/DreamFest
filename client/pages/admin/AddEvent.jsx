import React from 'react'
import { useHistory } from 'react-router-dom'

import { addEvent } from './addEventHelper'

import EventForm from '../../components/EventForm'

export default function AddEvent (props) {
  const history = useHistory()

  function submitEvent (event) {
    addEvent(event, history.push)
  }

  return (
    <EventForm
      action='Create Event'
      submitEvent={submitEvent}
    />
  )
}
