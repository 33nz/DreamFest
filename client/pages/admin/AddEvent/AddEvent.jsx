import React from 'react'
import { useHistory } from 'react-router-dom'

import { addEvent } from './addEventHelper'

import EventForm from '../../../components/events/EventForm/EventForm'

export default function AddEvent () {
  const history = useHistory()

  function submitEvent (event) {
    addEvent(event, history.push)
  }
  const initialState = {
    title: '',
    date: '',
    volunteersNeeded: 0,
    description: ''
  }
  return (
    <EventForm
      formData={initialState}
      action='Create Event'
      submitEvent={submitEvent}
    />
  )
}
