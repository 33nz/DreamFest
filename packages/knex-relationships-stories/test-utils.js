const { within } = require('@testing-library/dom')
const { JSDOM } = require('jsdom')

const render = (response) => within(new JSDOM(response.text).window.document)

module.exports = { render }
