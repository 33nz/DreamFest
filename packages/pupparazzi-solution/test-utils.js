const { within } = require('@testing-library/dom')
const { JSDOM } = require('jsdom')

export const render = (response) =>
  within(new JSDOM(response.text).window.document)
