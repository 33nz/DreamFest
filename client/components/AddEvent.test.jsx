import React from 'react'
import { fireEvent, waitFor } from '@testing-library/react'
import AddEvent from './AddEvent'
import { renderWithRedux } from '../test-utils'

jest.mock('../api/events', () => {
  return {
    addEvent: () => Promise.resolve()
  }
})

describe('Adding a new event', () => {
  it('Redirects to /garden', () => {
    const fakeHistory = []
    const { getByRole } = renderWithRedux(<AddEvent history={fakeHistory} />)
    const addButton = getByRole('button')
    fireEvent.click(addButton)
    return waitFor(() => expect(fakeHistory).toHaveLength(1))
      .then(() => expect(fakeHistory[0]).toBe('/garden'))
  })
})
