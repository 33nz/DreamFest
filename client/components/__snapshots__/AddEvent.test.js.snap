// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="create-event-form columns is-8"
>
  <div
    className="column"
  >
    <h1>
      Kelmarna Gardens
    </h1>
    <h3>
      Create new event
    </h3>
    <form
      onSubmit={[Function]}
    >
      <h5>
        Event Title
      </h5>
      <input
        className="input is-normal"
        name="title"
        onChange={[Function]}
        placeholder="event title"
        type="text"
        value=""
      />
      <h5>
        Date
      </h5>
      <input
        className="input is-normal"
        name="date"
        onChange={[Function]}
        placeholder="date and time"
        type="datetime-local"
        value=""
      />
      <h5>
        Volunteers Needed
      </h5>
      <input
        className="input is-normal"
        name="volunteers"
        onChange={[Function]}
        placeholder="number of volunteers needed"
        type="number"
        value={0}
      />
      <h5>
        Event Description
      </h5>
      <textarea
        className="textarea is-normal"
        name="description"
        onChange={[Function]}
        placeholder="event description"
        value=""
      />
      <button
        className="button my-4 is-primary"
      >
        Add Event
      </button>
    </form>
  </div>
  <div
    className="event-preview column"
  >
    <h1>
      Event Preview
    </h1>
    <div
      className="box"
    >
      <p>
        Weeds worker bee
      </p>
      <p>
        Date
      </p>
      <p>
        volunteers
      </p>
      <p>
        Description here...
      </p>
    </div>
  </div>
</div>
`;
